overlord/snapstate/snapmgr.go

```diff
@@ -420,6 +435,9 @@ func (m *SnapManager) doDiscardSnap(t *state.Task, _ *tomb.Tomb) error {
 			newSeq = append(newSeq, si)
 		}
 		snapst.Sequence = newSeq
+		if snapst.Current == ss.Revision {
```

Comment:
indeed if we used snapst.Current Unset to mean inactive (and discared the active flag), and just checked for len(Sequence) for other cases, this bit of code could go away

we would have though a concept of Current quite anymore though, only Active, which means some helper/cases might need some thinking